Lib/ldapurl.py

```diff
@@ -130,58 +132,64 @@ def __ne__(self,other):
     return not self.__eq__(other)
 
 
-class LDAPUrlExtensions(UserDict):
-  """
-  Models a collection of LDAP URL extensions as
-  dictionary type
-  """
-
-  def __init__(self,default=None):
-    UserDict.__init__(self)
-    for k,v in (default or {}).items():
-      self[k]=v
-
-  def __setitem__(self,name,value):
+class LDAPUrlExtensions(MutableMapping):
```

Comment:
Please add ``__slots__ = ('_data',)`` to avoid an additional ``__dict__`` on every instance.

Reply:
This looks addressed already but isn't marked as such yet, so the PR is still sitting with "changes requested" on it.